client/src/components/NavigationDrawer/NavigationDrawer.tsx

```diff
@@ -0,0 +1,155 @@
+    for (let i = 0; i < files.length; i++)
+      formData.append("outline_files", files[i]);
+
+    axios.post("/files", formData, {
+      headers: { "Content-Type": "multipart/form-data" }
+    })
+      .then(res => {
+        // if they share a key, rename the new one
+        for (const key in res.data) {
+          if (key in data) {
+            let i = 1;
+            while (`${key} (${i})` in data)
+              i++;
+            res.data[`${key} (${i})`] = res.data[key];
+            delete res.data[key];
+          }
+        }
+
+        setData({ ...data, ...res.data });
+      })
+      .catch((error) => {
+        console.log(error);
+      }).finally(() => {
+        setLoading([]);
+      });
+  };
+
+  return (
+    <div className="flex flex-col w-full md:p-4 p-0 bg-surface">
+      <p className="m-5 ml-2 font-bold">Courses</p>
+
+      {data && Object.entries(data).map(([course, courseData]) => (
+        <Button
+          variant="text"
+          to={`/review/${urlFormat(course)}`}
+          key={course}
+          className={classnames(
+            "text-gray-900",
+            "mt-2",
+            "flex",
+            "flex-row",
+            "p-4",
+            currentCourse === course && "bg-primary-90",
+          )}
+        >
+          <span className="material-icons text-gray-600 text-base flex items-center justify-center">
+            {["circle", "square", "pentagon"][Math.abs(course.split("").reduce((a, b) => a + b.charCodeAt(0), 0)) % 3]}
+          </span>
+          <div className="flex flex-col ml-2 min-w-0">
+            <p className="flex items-center font-bold">
+              {course}
+            </p>
+            <p className={classnames("truncate", "md:hidden")}>
+              {courseData.topic} - The course.course descriptions can be quite long, so we truncate them to save space.
+            </p>
+          </div>
+          <p className="ml-auto flex items-center justify-center">
+            {courseData.assessments.length}
+          </p>
+          <span className="material-icons text-gray-600 flex items-center justify-center block md:hidden">
+            arrow_right
+          </span>
+        </Button>
+      ))}
+      {loading.length > 0 && (
+        <div className="flex flex-col w-full">
+          {loading.map((file) => (
+            <Button
+              variant="text"
+              key={file}
+              className={classnames("text-gray-900", "mt-2", "flex", "flex-row", "p-4", "opacity-50", "pointer-events-none",
+              )}>
+              <span className="material-icons text-gray-600 text-base flex items-center justify-center animate-spin">
+                autorenew
+              </span>
+              <div className="flex flex-col ml-2 min-w-0">
+                <p className={classnames("truncate")}>
+                  {file}
+                </p>
+              </div>
+            </Button>
+          ))}
+        </div>
+      )}
+      <Button
```

Comment:
This button is implemented more complex that it should be, in terms of design/css.

Reply:
disabled prop 👍